src/main/java/de/bonndan/nivio/output/map/hex/Hex.java

```diff
@@ -103,11 +120,19 @@ public static Hex of(long x, long y, float scaling) {
         return new Hex(qi, ri, si);
     }
 
-
+    /**
+     * Returns the distance to the target hex in number of tiles.
+     *
+     * @param target target hex
+     * @return number of tiles
+     */
     public int distance(Hex target) {
-        return lengths(this.subtract(target));
+        Hex hex = this.subtract(target);
+        double l = (Math.abs(hex.q) + Math.abs(hex.r) + Math.abs(hex.s)) / 2.0;
```

Comment:
Why is the distance the sum of all coordinates divided by 2?

Reply:
https://www.redblobgames.com/grids/hexagons/#distances I will add this link as comment, too.